roles/platform/tasks/teardown_aws_terraform_authz.yml

```diff
@@ -0,0 +1,78 @@
+---
+
+# Copyright 2021 Cloudera, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- name: Remove CDP Cross Account Credential for AWS
+  when: plat__teardown_deletes_credential
+  cloudera.cloud.env_cred:
+    name: "{{ plat__xacccount_credential_name }}"  # TODO: Make specific to AWS Teardown as credentials can be for multiple environments
+    state: absent
+
+- name: Check if Terraform workspace directory exists
+  ansible.builtin.stat: 
+    path: "{{ plat__terraform_template_dir }}/workspace/plat"
+  register: workdir
+
+- name: Ensure the Terraform workspace directory exists
+  ansible.builtin.copy:
+    src: "{{ plat__terraform_template_dir }}/plat/"
+    dest: "{{ plat__terraform_workspace_dir }}/workspace/plat"
+  # when: not workdir.stat.exists # TODO: This won't override if any changes to .tf are made. Good or Bad?
+
+- name: Prompt to debug Terraform deployment 
+  ansible.builtin.pause:
+    prompt: "Ready to do Terraform destroy on role files in {{ plat__terraform_workspace_dir }}/workspace/plat"
+  when: debug_terraform | default(false) | bool
+
+- name: Destroy Terraform authz resources
+  community.general.terraform:
+    project_path: "{{ plat__terraform_workspace_dir }}/workspace/plat"
+    state: "absent"
+    force_init: yes
+  register: tf_result
+  retries: 3
+  delay: 10
+  until: tf_result is succeeded
+
+- name: Prompt to debug Terraform deployment 
+  ansible.builtin.pause:
+    prompt: "Terraform destroy on role files complete. Check AWS Console."
+  when: debug_terraform | default(false) | bool
+
+- name: Remove the Terraform template directory
+  ansible.builtin.file:
+    path: "{{ plat__terraform_template_dir }}/plat/"
+    state: absent
+
+- name: Remove the Terraform workspace directory
+  ansible.builtin.file:
+    path: "{{ plat__terraform_workspace_dir }}/workspace/plat"
+    state: absent
+    
+# TODO: Figure out how to handle below conditional with Terraform
```

Comment:
TBD?

Reply:
Yes I need to figure out how to skip teardown of Terraform-created resources if some flags (e.g. env.teardown.delete_cross_account: False) are specified. I've removed the TODO from the code and will create an issue for these tasks.